flow/layers/aiks_layer.cc

```diff
@@ -0,0 +1,67 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/aiks_layer.h"
+
+#include <utility>
+
+namespace flutter {
+
+AiksLayer::AiksLayer(const SkPoint& offset,
+                     const std::shared_ptr<const impeller::Picture>& picture)
+    : offset_(offset), picture_(picture) {
+#if IMPELLER_SUPPORTS_RENDERING
+  if (picture_) {
+    auto bounds = picture_->pass->GetElementsCoverage(std::nullopt)
```

Comment:
Thie one is also really slow.

Reply:
Looks like the TextFrame bounds are slow, at least in ComplexLayout (This is UI thread)

![image](https://github.com/flutter/engine/assets/8975114/e787d915-89b7-4771-8d6d-09a9f64cf9eb)

Reply:
We end up needing to get these multiple times per frame AFAICT, we should probably cache them once we've calculated them.

Reply:
See also: https://github.com/flutter/flutter/issues/133388 . We should just cache the value we get from Skia. They're computing the bounds anyway.

Reply:
Not just the text bounds - why isn't the picture itself remembering its own bounds? These will get used frame after frame and we have to iterate the elements each time to save ... 16 bytes in the object?

Reply:
This is where we could just ask the rtree for the bounds :)

Reply:
Does that code always produce an rtree? Then, yes, it's a very cheap way to get the bounds. We had been making that mistake in DL for a long time until I fixed it recently in the "split DL builder" PR.